lib/saxy/parser/builder.ex

```diff
@@ -889,12 +889,9 @@ defmodule Saxy.Parser.Builder do
           token in unquote(edge_ngrams("]]")) when more? ->
             halt!(element_cdata(token, more?, original, pos, state, len))
 
-          char <> rest when is_ascii(char) ->
+          char <> rest ->
             element_cdata(rest, more?, original, pos, state, len + 1)
 
-          <<codepoint::utf8>> <> rest ->
-            element_cdata(rest, more?, original, pos, state, len + Utils.compute_char_len(codepoint))
```

Comment:
I think we can the same way how dangling UTF-8 fragments is handled 

For example https://github.com/qcam/saxy/blob/master/lib/saxy/parser/builder.ex#L540-L541